### PR TITLE
Story 1 - complete

### PR DESCRIPTION
**Pull request by GenadiKozarev**
please review : )

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$query = $db->prepare('SELECT * FROM `gwent_cards`;');
+$query->execute();
+
+$allCards = $query->fetchAll();
+
+/*uncomment if you want to check the whole db array
+print_r($allCards);
+*/
```
Comment (norayna) [inline index.php]: i can see why this is handy but it shouldn't really be in the repo - commented out code in general is not the best to have in a repo

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            '<span>________________________________________</span>' .
+            '<h2>' . $card['name'] . '</h2>' .
+            '<img src="' . $card['image_link'] . '" alt="an image of the card">' .
+            '<p>Unit Strength: ' . $card['unit_strength'] . '</p>' .
+            '<p>Type: ' . $card['type'] . '</p>' .
+            '<p>Price: ' . $card['price'] . '</p>' .
+            '<p>Territory: ' . $card['territory'] . '</p>';
+    }
+    if($allCards === [['name' => 3.14, 'image_link' => 3.14, 'unit_strength' => 'PIE', 'type' => 3.14, 'price' => 'PIE', 'territory' => 3.14]]) {
+        return 'PIE IN YOUR FACE';
```
Comment (norayna) [inline functions.php]: this is saying that if the function receives this _exact_ incorrect input, then it returns an error message. it doesn't take care of _all_ possible incorrect inputs, it just chooses the specific one that happens to be in the unit test. 

in the future, someone might edit your code and, for example, accidentally pass the wrong array into the function. your function would end up throwing an error that would be displayed on the page plainly for users to see.

you need to handle all possible invalid arrays (valid arrays should have all the keys you need to create the html).

Comment (norayna) [reply functions.php]: also unfortunately you need to put a more informative (boring) error message

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+function displayCollection(array $allCards) : string {
+    $collection = '';
+    foreach ($allCards as $card) {
+        $collection .=
+            '<span>________________________________________</span>' .
```
Comment (norayna) [inline functions.php]: this should be achieved using css instead (or removed)

**Review by norayna — CHANGES_REQUESTED** [review]
changes requested

File: index.php (hunk 1 above)
Comment (GenadiKozarev) [reply index.php]: removed

File: functions.php (hunk 2 above)
Comment (GenadiKozarev) [reply functions.php]: fixed

File: functions.php (hunk 3 above)
Comment (GenadiKozarev) [reply functions.php]: fixed

File: formValidation.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    $addCardQuery->bindParam(':newCardName', $newCardName);
+    $addCardQuery->bindParam(':newCardImageUrl', $newImageUrl);
+    $addCardQuery->bindParam(':newCardUnitStrength',$newUnitStrength);
+    $addCardQuery->bindParam(':newCardType', $newCardType);
+    $addCardQuery->bindParam(':newCardPrice', $newCardPrice);
+    $addCardQuery->bindParam(':newCardTerritory', $newCardTerritory);
+
+    $addCardQuery->execute();
+    header('Location: index.php');
+}
```
Comment (norayna) [inline formValidation.php:27]: this is from story 2?

File: index.php [hunk 5]
```diff
@@ -37,9 +29,20 @@
...
+        <form action="formValidation.php" method="POST">
+            <input type="text" placeholder="Enter card name" name="cardName" required>
+            <input type="text" placeholder="Enter image URL" name="imageUrl" required>
+            <input type="text" placeholder="Enter unit strength" name="unitStrength" required>
+            <input type="text" placeholder="Enter type" name="cardType" required>
+            <input type="text" placeholder="Enter price" name="cardPrice" required>
+            <input type="text" placeholder="Enter territory" name="cardTerritory" required>
+            <input type="submit" name="submit" value="Add a new card">
+        </form>
+    </div>
```
Comment (norayna) [inline index.php:46]: this is also from story 2?

**Review by norayna — CHANGES_REQUESTED** [review]
all good except some code from story 2 is here

File: index.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Marcellus&family=Suez+One&display=swap" rel="stylesheet">
+</head>
+
+
+<body>
+    <h1>Gwent cards collection</h1>
+    <h2>add a new card to your collection</h2>
```
Comment (norayna) [inline index.php:32]: this heading is at the top of the page but the form to add a new card is at the bottom of the page - either move heading to the bottom or form to the top so they are together

File: index.php (hunk 5 above)
Comment (norayna) [reply index.php:46]: make sure to pull this onto the story 2 branch

**Review by norayna — APPROVED** [review]

**Pull request merged**